plugins/de.fraunhofer.ipa.ros.observer.generator/src/de/fraunhofer/ipa/ros/observer/generator/ObserverPyCodeGenerator.xtend

```diff
@@ -0,0 +1,85 @@
+package de.fraunhofer.ipa.ros.observer.generator
+
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.generator.AbstractGenerator
+import org.eclipse.xtext.generator.IFileSystemAccess2
+import org.eclipse.xtext.generator.IGeneratorContext
+import org.eclipse.xtext.generator.IOutputConfigurationProvider
+import org.eclipse.xtext.generator.OutputConfiguration
+import java.util.Set
+import ros.Node
+
+class CustomOutputProvider implements IOutputConfigurationProvider {
+	public final static String OBSERVER_OUTPUT = "OBSERVER_OUTPUT"
+	
+	override Set<OutputConfiguration> getOutputConfigurations() {
+		var OutputConfiguration observer_config = new OutputConfiguration(OBSERVER_OUTPUT)
+		observer_config.setDescription("OBSERVER_OUTPUT");
+		observer_config.setOutputDirectory("./src-gen/observers/");
+		observer_config.setOverrideExistingResources(true);
+		observer_config.setCreateOutputDirectory(true);
+		observer_config.setCleanUpDerivedResources(true);
+		observer_config.setSetDerivedProperty(true);
+		return newHashSet(observer_config)
+	}
+}
+
+class ObserverPyCodeGenerator extends AbstractGenerator {
+
+	int count_sub
+
+	override void doGenerate(Resource resource, IFileSystemAccess2 fsa, IGeneratorContext context) {
+			for (node : resource.allContents.toIterable.filter(Node)){
+				fsa.generateFile(node.getName()+".py",CustomOutputProvider::OBSERVER_OUTPUT,node.compile)
+				}
+			}
+
+def compile(Node node) {
+	count_sub = node.subscriber.size
+'''
+from rosgraph_monitor.observer import TopicObserver
+«FOR sub:node.subscriber»
+from «sub.message.package.name».msg import «sub.message.name»
+«ENDFOR»
+from diagnostic_msgs.msg import DiagnosticStatus, KeyValue
+
+
+class QualityObserver(TopicObserver):
+    def __init__(self, name):
+    	
+        topics = [
+«FOR sub:node.subscriber»«val count_sub=count_sub--»
+        ("«sub.name»", «sub.message.name»)«IF count_sub > 1 »,«ENDIF»«ENDFOR»
+        ]
+
+        super(QualityObserver, self).__init__(
+            name, 10, topics)
+
+    def calculate_attr(self, msgs):
+        status_msg = DiagnosticStatus()
+        
+        #Write here your code (or uncomment the following lines)
+        #if len(msgs) < 2:
+        #    print("Incorrect number of messages")
+        #    return status_msg
+        
+        #if not isinstance(msgs[0], Int32) or not isinstance(msgs[1], Int32):
+        #    print("Incorrect instance of message")
+        #   return status_msg
```

Comment:
Could you please delete these lines? I am taking care of it in [observer.py](https://github.com/ipa-hsd/rosgraph_monitor/blob/87a415f0ca4da08bfd829ba84cf5fa567888deb2/src/rosgraph_monitor/observer.py#L105-L111). And since it is hard-coded (len > 2 & type == `Int32`) it might cause the code to crash, as in case of [this](https://github.com/rosin-project/rosgraph_monitor/blob/observers/src/rosgraph_monitor/observers/quality_observer.py)

Reply:
Done! Thanks!